### PR TITLE
Fix popup tab loading interruption

### PR DESCRIPTION
**Pull request by mallexxx**
Task/Issue URL: https://app.asana.com/0/414709148257752/1202193006784955/f
GH Issue URL: https://github.com/duckduckgo/iOS/issues/1202
Tech Design URL:
CC: @miasma13 

**Description**:
- Fixes loading interruption when WebView created in webView:createWebViewWithConfiguration:..
- Loading is interrupted for cleaned urls but managed by a creator tab otherwise
- This fixes an issue of Printing not working on a new tab created for printing

**Steps to test this PR**:
1. Validate print works on https://www.theenglishkitchen.co/2009/09/beer-battered-fish.html?m=1
2. Validate links opened in new tab work (e.g. in yandex search)
3. Validate pop-up windows created with js work

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**

### CODE REVIEW

File: DuckDuckGo/TabViewController.swift [hunk 1]
```diff
@@ -1630,6 +1637,7 @@ extension TabViewController: WKUIDelegate {
             completionHandler(nil)
         }
     }
+
```
Comment (miasma13) [inline DuckDuckGo/TabViewController.swift]: Unnecessary edit.

File: DuckDuckGo/TabViewController.swift [hunk 2]
```diff
@@ -349,14 +352,18 @@ class TabViewController: UIViewController {
...
-                                           onFinishExtracting: { },
-                                           completion: { [weak self] cleanURL in self?.load(urlRequest: .userInitiated(cleanURL)) })
-            } else {
-                load(urlRequest: request)
+        } else if let url = request?.url {
+            var loadingStopped = false
+            linkProtection.getCleanURL(from: url) { [weak self] in
+                if loadedByParentTab {
+                    loadingStopped = true
+                    self?.webView.stopLoading()
```
Comment (miasma13) [inline DuckDuckGo/TabViewController.swift]: Can you please provide a bit explanation for this logic?

File: DuckDuckGo/TabViewController.swift [hunk 3]
```diff
@@ -349,14 +352,18 @@ class TabViewController: UIViewController {
...
+                }
+                showProgressIndicator()
+            } onFinishExtracting: {
+            } completion: { [weak self] cleanURL in
+                guard url != cleanURL || !loadedByParentTab || loadingStopped else { return }
```
Comment (miasma13) [inline DuckDuckGo/TabViewController.swift]: What is purpose for this guard?
Does it even makes sense, as it seems that both `loadedByParentTab` and `loadingStopped` will always have same values.

**Review by miasma13 — CHANGES_REQUESTED** [review]

File: DuckDuckGo/TabViewController.swift (hunk 3 above)
Comment (mallexxx) [reply DuckDuckGo/TabViewController.swift]: the check here passes if either loading has stopped or if the tab is NOT loaded by its parent, done so not to interrupt an ongoing popup request loading if possible

**Comment by mallexxx** [comment]
Thanks, @miasma13, I've changed naming and added some comments for clarity
TLDR: the popup window based printing breaks because we‘ve been stopping the popup loading (initiated by its parent js) when URL cleanup happened. This change tries to avoid loading breakage by stopping loading only for non-parent-initiated requests or if the URL was modified. Otherwise (for a popup) we will just keep current request going

**Review by miasma13 — APPROVED** [review]
Thanks for the explanation
